src/filesystem.c

```diff
@@ -161,6 +162,24 @@ rcutils_join_path(
   return rcutils_format_string(allocator, "%s%s%s", left_hand_path, delimiter, right_hand_path);
 }
 
+char *
+rcutils_norm_path(
+  const char * path,
+  rcutils_allocator_t allocator)
+{
+  if (NULL == path) {
+    return NULL;
+  }
+
+#ifdef  _WIN32
+  const char * delimiter = "\\";
+#else
+  const char * delimiter = "/";
+#endif  // _WIN32
```

Comment:
Not sure if we should duplicate this in every function using it.
Is it possible to define those at the top of the file to something like `RCUTILS_PATH_DELIMITER` and reuse it in all functions using it instead?

Reply:
The top of the header file or the c file?

Reply:
If we don't plan on exporting it to users (which I think we don't have a need for right now) I'd say yeah put it in the c file for now.